types/openapi-factory/index.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for openapi-factory 4.2
+// Project: https://github.com/wparad/openapi-factory.js
+// Definitions by: Daan Boerlage <https://github.com/runebaas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare namespace OpenApi {
+    interface OpenApiOptions {
+        requestMiddleware?: () => any;
+        responseMiddleware?: () => any;
+        errorMiddleware?: () => any;
+    }
+}
+
+declare class OpenApi {
+    constructor(options: OpenApi.OpenApiOptions, overrideLogger?: () => void);
+
+    setAuthorizer(authorizerFunc: (req?: any) => any): void;
+    onEvent(onEventFunc: (req?: any) => any): void;
+    onSchedule(onScheduleFunc: (req?: any) => any): void;
+
+    head(route: string, p0: (req?: any) => any): void;
```

Comment:
is there an option to do something like:
`head(route: string, p0: (req?: object) => Promise<any>): void`